packages/python-google-compute-engine/google_compute_engine/distro_lib/helpers.py

```diff
@@ -42,6 +42,27 @@ def CallDhclient(
     logger.warning('Could not enable interfaces %s.', interfaces)
 
 
+def CallDhclientIpv6(interfaces, logger, dhclient_script=None):
+  """Configure the network interfaces for IPv6 using dhclient.
+
+  Args:
+    interface: string, the output device names for enabling IPv6.
+    logger: logger object, used to write to SysLog and serial port.
+    dhclient_script: string, the path to a dhclient script used by dhclient.
+  """
+  logger.info('Enabling IPv6 on the Ethernet interfaces %s.', interfaces)
+
+  dhclient_command = ['dhclient']
+
+  if dhclient_script and os.path.exists(dhclient_script):
+    dhclient_command += ['-sf', dhclient_script]
+
+  try:
+    subprocess.check_call(dhclient_command + ['-1', '-6', '-v'] + interfaces)
```

Comment:
why these args?

Reply:
OK, looked into the issue driving this change. Notes:
`-6` is obviously needed. 
`-v` is quite noisy, i think since we are wrapping these system commands we should drop it.
`-1` has a behavior change between IPv4 and IPv6, where in the latter case it forces the refresh to limit itself to the timeout setting from dhclient.conf. However for some systems (e.g. ubuntu 1804), this is set to 5 minutes. In any DHCP failure case, that will hang the entire network daemon for five minutes.
Recommend running this in the background to protect against that case and possibly adding some checks to see if we have any running before doing this.

Reply:
ah and `-x` from our normal dhclient calls is not needed, that only kills established dhclient daemons

Reply:
the timeout can be easily enforced by hand by prepending the command with the 'timeout' comamnd, i.e.:

timeout dhclient -1 -6 eth0

The timeout command is part of the coreutils, so its available in all fedora/debian/etc based systems by default.

Reply:
Another alternative is to fork and run the command as a backgrounded process. I'm not sure which approach is better. @zmarano @hopkiw any preferences?

Reply:
FWIW if you choose to background the process some care may be needed to make sure you don't fork too many such clients in cases where the interface goes up and down. you could try to kill previous processes to make sure you only run one at a time, but then you could worry about how to kill the "correct" dhclient since the user may be running instances of their own. For that you could write a pid file, etc..

You could also choose to ignore these nuances and hope we simply dont run into these corner cases. Not sure what level of paranoia is warranted here.

Reply:
for me personally a simple safeguard like timeout or setting the timeout argument to subprocess.check_call() would be reasonable. as i understand it, this 5m hang condition would be reached any time dhcp fails for a secondary interface, and the attempts are triggered by a metadata key.